src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -332,7 +339,52 @@ async def create_docker_build_context(
     else:
         logger.debug("Did not build any packages for Docker image")
 
-    embedded_pkgs_digest = [built_package.digest for built_package in embedded_pkgs]
+    # Create digests for embedded packages. For upstream Docker images, we use only the image ID
+    # to ensure hash stability. This prevents changes in image tags (which may include timestamps
+    # or other dynamic values) from affecting the build context hash of dependent images.
+    embedded_pkgs_digest = []
+
+    # For Docker images, we need to extract the metadata filename and create stable digests
+    docker_metadata_gets = []
+    docker_package_indices = []
+    for i, built_package in enumerate(embedded_pkgs):
+        field_set = pkgs_wanting_embedding[i]
+        if request.build_upstream_images and isinstance(
+            getattr(field_set, "source", None), DockerImageSourceField
+        ):
+            docker_metadata_gets.append(Get(DigestContents, Digest, built_package.digest))
```

Comment:
We are moving off `Get/MultiGet` in favor of call-by-name. Sorry, this isn't properly documented yet. See https://github.com/pantsbuild/pants/discussions/18905 for context.

So instead of `Get(DigestContents, Digest, built_package.digest)` you'd use `get_digest_contents(get_digest_contents)` (with `from pants.engine.intrinsics import get_digest_contents`)

and then you'd replace `MultiGet` with `concurrently`, imported from `pants.engine.internals.selectors`.